agent/consul/leader_connect.go

```diff
@@ -567,6 +567,17 @@ func (s *Server) startConnectLeader() {
 
 // stopConnectLeader stops connect specific leader functions.
 func (s *Server) stopConnectLeader() {
+	s.caProviderReconfigurationLock.Lock()
+	defer s.caProviderReconfigurationLock.Unlock()
```

Comment:
Should you scope this lock/unlock just around the new provider conditional block, rather than extending it over all of the leader routine operations as it is here?